### ISSUE TITLE
Table of Contents inaccessible to voiceover

### ISSUE THREAD

**Issue by avaseghi** [wenqili/adjacent#15]
The table of contents cannot be read by voiceover.

**Comment by wenqili** [comment]
- [x] Refactor catalog, add open/close button
- [x] Add tabindex and button role to catalog
